portals/knowledge-management/src/Components/MobileMenu.vue

```diff
@@ -0,0 +1,130 @@
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+-->
+<template>
+    <div>
+        <Bars3Icon @click="openDrawer" class="h-6 w-6 text-gray-700" aria-hidden="true" />
+
+        <Transition name="backdrop">
+            <div v-if="visible" class="fixed inset-0 z-40 bg-black bg-opacity-50" @click="closeDrawer"></div>
+        </Transition>
+
+        <Transition name="drawer">
+            <div v-if="visible" class="fixed top-0 right-0 w-64 h-full bg-white shadow-xl z-50" @click.stop>
+                <div class="flex justify-between items-center px-4 py-3 border-b">
+                    <h3 class="text-lg font-semibold text-gray-700">Menu</h3>
+                    <XMarkIcon @click="closeDrawer" class="h-5 w-5 text-gray-700" />
+                </div>
+
+                <div class="p-4 space-y-5">
+                    <template v-for="item in visibleMenuItems" :key="item.label">
+                        <router-link
+                            :to="{ name: item.routeName }"
+                            custom
+                            v-slot="{ navigate, isActive, isExactActive }"
+                        >
+                            <a
+                                @click="navigate"
+                                class="flex items-center font-medium text-sm"
+                                :class="[
+                                    isActive || isExactActive ? 'text-brand-500' : 'text-gray-700',
+                                    'hover:text-brand-500',
+                                ]"
+                            >
+                                {{ item.label }}
+                            </a>
+                        </router-link>
+                    </template>
+                </div>
+            </div>
+        </Transition>
+    </div>
+</template>
+
+<script setup>
```

Comment:
Precident says that Vue files should start with the Script then have the template below it. Let's match that here.